test/math.jl

```diff
@@ -983,8 +983,68 @@ end
 
 isdefined(Main, :Furlongs) || @eval Main include("testhelpers/Furlongs.jl")
 using .Main.Furlongs
-@test hypot(Furlong(0), Furlong(0)) == Furlong(0.0)
-@test hypot(Furlong(3), Furlong(4)) == Furlong(5.0)
-@test hypot(Furlong(NaN), Furlong(Inf)) == Furlong(Inf)
-@test hypot(Furlong(Inf), Furlong(NaN)) == Furlong(Inf)
-@test hypot(Furlong(Inf), Furlong(Inf)) == Furlong(Inf)
+@test (@inferred hypot(Furlong(0), Furlong(0))) == Furlong(0.0)
+@test (@inferred hypot(Furlong(3), Furlong(4))) == Furlong(5.0)
+@test (@inferred hypot(Furlong(NaN), Furlong(Inf))) == Furlong(Inf)
+@test (@inferred hypot(Furlong(Inf), Furlong(NaN))) == Furlong(Inf)
+@test (@inferred hypot(Furlong(0), Furlong(0), Furlong(0))) == Furlong(0.0)
+@test (@inferred hypot(Furlong(Inf), Furlong(Inf))) == Furlong(Inf)
+@test (@inferred hypot(Furlong(1), Furlong(1), Furlong(1))) == Furlong(sqrt(3))
+# @test (@inferred hypot(Furlong(Inf), Furlong(NaN), Furlong(0))) == Furlong(Inf)
+@test (@inferred hypot(Furlong(Inf), Furlong(Inf), Furlong(Inf))) == Furlong(Inf)
+@test isnan(hypot(Furlong(NaN), Furlong(0), Furlong(1)))
+
+@testset "hypot" begin
+    # @test (@inferred hypot(floatmax())) == floatmax()
+    @test (@inferred hypot(floatmax(), floatmax())) == Inf
+    @test (@inferred hypot(floatmin(), floatmin())) == √2floatmin()
+    # @test (@inferred hypot(floatmin(), floatmin(), floatmin())) == √3floatmin()
+    # @test (@inferred hypot(1e-162)) ≈ 1e-162
+    # @test (@inferred hypot(2e-162, 1e-162, 1e-162)) ≈ hypot(2,1,1)*1e-162
+    # @test (@inferred hypot(1e162)) ≈ 1e162
+    @test hypot(-2) === 2.0
+    @test hypot(-2, 0) === 2.0
+    let i = typemax(Int)
+        @test (@inferred hypot(i, i)) ≈ i * √2
+        # @test (@inferred hypot(i, i, i)) ≈ i * √3
+        # @test (@inferred hypot(i, i, i, i)) ≈ 2.0i
+        # @test (@inferred hypot(i//1, 1//i, 1//i)) ≈ i
+    end
+    let i = typemin(Int)
+        @test (@inferred hypot(i, i)) ≈ √2i
```

Comment:
I'm confused by this:
```julia
julia> hypot(typemin(Int), typemin(Int))
-1.3043817825332783e19
```
Why is this correct?

Reply:
@simonbyrne ^ (I need a bat signal for you)

Reply:
That's a bug: `hypot(x::T, y::T) where T<:Number` calls `abs` and assumes it gives a positive result.  I commented on the same problem here: https://github.com/JuliaLang/julia/pull/30301#issuecomment-447145220

Reply:
> calls `abs` and assumes it gives a positive result

I understand what's the problem with `abs(typemin(Int))`, but I'd say that in general this is a good assumption :sweat_smile:

Ok, I'm going to mark this and the other tests that are consistently failing as broken